botocore/credentials.py

```diff
@@ -2170,6 +2171,7 @@ def load_credentials(self):
             logger.debug("Looking for credentials via: %s", provider.METHOD)
             creds = provider.load()
             if creds is not None:
+                self.check_and_register_feature_id(provider.METHOD)
```

Comment:
You may have a better understanding of this than me, but I think the`load_credentials` method here is ultimately called from session.create_client, then those are cached and reused for multiple service calls with the same client.  

That means this feature id will only show up in the first call using a client, right?  Is that the desired behavior, or is there something I'm missing here?